packages/core-data/src/private-actions.js

```diff
@@ -14,3 +24,102 @@ export function receiveRegisteredPostMeta( postType, registeredPostMeta ) {
 		registeredPostMeta,
 	};
 }
+
+/**
+ * Duplicates an entity record and, optionally, modifies it.
+ *
+ * @param {string}   kind                    Entity kind.
+ * @param {string}   name                    Entity name.
+ * @param {string}   recordId                Entity record ID.
+ * @param {Object}   edits                   Edits to apply to the record.
+ * @param {Object}   options                 Options object.
+ * @param {Function} options.__unstableFetch Custom fetch function.
+ * @param {boolean}  options.throwOnError    Whether to throw an error if the request fails.
+ *
+ * @return {Promise} Promise resolving to the updated record.
+ */
+export const duplicateEntityRecord =
```

Comment:
This doesn't really describe what's happening in the attachments endpoint, but it leaves things open for other endpoints to support duplication.